forte/data/data_pack.py

```diff
@@ -244,14 +252,14 @@ def _validate(self, entry: EntryType) -> bool:
         return isinstance(entry, SinglePackEntries)
 
     @property
-    def text(self) -> str:
+    def text(self, text_payload_index=0) -> str:
         r"""Return the text of the data pack"""
-        return self._text
+        return self.get_single(TextPayload, text_payload_index).cache
```

Comment:
I think you are misusing `get_single` here. `get_single` is used to get one object of a particular type, for example, `pack.get(Article)`.  This `get_single` is used when there is one entry of this type, or you want to take the first one for demo purposes.

In a `DataPack`, there should be multiple `TextPayload`, so I don't think you should do this.

Btw, we mentioned that we should move different types of `Payload` out and only use one function to access these `Payload`.

Reply:
Do you mean `image1 = pack.get_payload_at(1)`? I think this operation could be problematic as user needs to know the index of modality that he wants. Or do you think `image1 = pack.get_payload_at("image", 1)` is a good interface?